ui/raidboss/timeline_parser.ts

```diff
@@ -1,11 +1,52 @@
 import { Lang } from '../../resources/languages';
+import logDefinitions, { LogDefinitionTypes } from '../../resources/netlog_defs';
+import { buildNetRegexForTrigger } from '../../resources/netregexes';
 import { UnreachableCode } from '../../resources/not_reached';
 import Regexes from '../../resources/regexes';
 import { translateRegex, translateText } from '../../resources/translations';
+import { NetParams } from '../../types/net_props';
 import { LooseTimelineTrigger, TriggerAutoConfig } from '../../types/trigger';
 
 import defaultOptions, { RaidbossOptions, TimelineConfig } from './raidboss_options';
 
+const isLogDefinitionTypes = (type: string): type is LogDefinitionTypes => {
+  return type in logDefinitions;
+};
+
+const isStringArray = (value: unknown[]): value is string[] => {
+  return value.find((v) => typeof v !== 'string') === undefined;
+};
+
+const isStringOrStringArray = (value: unknown): value is number[] | string[] => {
+  if (Array.isArray(value)) {
+    if (isStringArray(value))
+      return true;
+    return false;
+  } else if (!['string', 'number'].includes(typeof value))
+    return false;
+  return true;
+};
```

Comment:
```suggestion
const isStringOrStringArray = (value: unknown): value is string | string[] => {
  if (Array.isArray(value)) {
    if (isStringArray(value))
      return true;
    return false;
  } 
  return typeof value === 'string';
};
```

Is this what you meant here?